### PR TITLE
feat(assistants): Add assistant left side navigation

### PR DESCRIPTION
**Pull request by knajjars**
https://github.com/cohere-ai/cohere-toolkit/assets/19353218/f5f566d2-09b8-4406-9e19-172a1c1ddb27

**AI Description**

This PR introduces a new component for rendering an agent card and a list of agents, and updates the `Layout` component to use the new `AgentsSidePanel` component. It also adds a new utility function for generating colors from the Cohere color palette.

- **New Components**:
  - `AgentCard`: Renders an agent card with the agent's name and a colored icon. If the agent is a base agent, it displays the Coral logo.
  - `AgentsList`: Renders a list of agents, including the most recent agents and base agents.
- **Updated Components**:
  - `Layout`: Replaced the `leftElement` prop with the new `AgentsSidePanel` component.
- **Deleted Components**:
  - `AddAgentButton`: Removed the component responsible for rendering a button to add a new agent.
  - `BaseAgentButton`: Removed the component for rendering a button to navigate to the default knowledge agent page.
  - `LeftPanel`: Removed the component that rendered the left panel containing the base agent, a list of agents, and an add agent button.
- **Updated Utility Functions**:
  - `getCohereColor`: Added a new utility function to generate colors from the Cohere color palette. It accepts an optional `id` parameter to generate a constant color or a random color if no `id` is provided.
- **Updated State Management**:
  - `settingsSlice`: Added a new `isAgentsSidePanelOpen` property to manage the state of the agents side panel.
  - `persistedStore`: Updated the version of the `settings` state and added the `setIsAgentsSidePanelOpen` method to the `useSettingsStore` function.

### CODE REVIEW

File: src/interfaces/coral_web/src/components/Agents/AgentCard.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          <Text className="truncate">{name}</Text>
+        </Transition>
+        <Transition
+          as="div"
+          show={isExpanded && !isBaseAgent}
+          enter="transition-opacity duration-100 ease-in-out delay-300 duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+        >
+          <Menu>
```
Comment (wujessica) [inline src/interfaces/coral_web/src/components/Agents/AgentCard.tsx:61]: thoughts on updating `KebabMenu.tsx` to be able to be reused here?

File: src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  } = useSettingsStore();
+
+  const navigationItems: {
+    label: string;
+    icon: IconProps['name'];
+    href?: string;
+    onClick?: () => void;
+  }[] = [
+    { label: 'Create Assistant ', icon: 'add', href: '/agents/new' },
+    { label: 'Sign Out', icon: 'profile', onClick: () => void 0 },
```
Comment (wujessica) [inline src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx:27]: what is `void 0` :o

File: src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        })}
+      >
+        {isAgentsSidePanelOpen && (
+          <Link href="/" shallow>
+            <div className="mr-3 flex items-baseline">
+              <Logo hasCustomLogo={env.NEXT_PUBLIC_HAS_CUSTOM_LOGO === 'true'} />
+            </div>
+          </Link>
+        )}
+        <button
```
Comment (wujessica) [inline src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx:56]: lets use `IconButton`

File: src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            {href ? (
+              <Link className="flex w-full justify-center" href={href} shallow>
+                <Icon name={icon} className="text-secondary-900" />
+              </Link>
+            ) : (
+              <button className="w-full" onClick={onClick}>
+                <Icon name={icon} className="text-secondary-900" />
+              </button>
+            )}
+          </Tooltip>
```
Comment (wujessica) [inline src/interfaces/coral_web/src/components/Agents/AgentsSidePanel.tsx:85]: could we use `IconButton` here too?

File: src/interfaces/coral_web/src/utils/getCohereColor.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * @description Get a color from the Cohere color palette, when no index is provided, a random color is returned
+ * @param id - id for generating a constant color in the palette
+ * @returns color from the Cohere color palette
+ */
+export const getCohereColor = (id?: string) => {
+  if (id === undefined) {
+    const randomIndex = Math.floor(Math.random() * COLOR_LIST.length);
+    console.log(randomIndex);
```
Comment (wujessica) [inline src/interfaces/coral_web/src/utils/getCohereColor.ts:18]:
```suggestion
```

File: src/interfaces/coral_web/src/utils/getCohereColor.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    return COLOR_LIST[randomIndex];
+  }
+
+  const index = id.charCodeAt(0) % COLOR_LIST.length;
+  return COLOR_LIST[index % COLOR_LIST.length];
```
Comment (wujessica) [inline src/interfaces/coral_web/src/utils/getCohereColor.ts:23]: do we need to mod again? 🤔

**Review by wujessica — APPROVED** [review]

File: src/interfaces/coral_web/src/utils/getCohereColor.ts (hunk 6 above)
Comment (knajjars) [reply src/interfaces/coral_web/src/utils/getCohereColor.ts:23]: whoops! removed

File: src/interfaces/coral_web/src/components/Agents/AgentCard.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/coral_web/src/components/Agents/AgentCard.tsx]: good idea!

**Pull request merged**